shared/src/main/scala/tastyquery/reader/classfiles/JavaSignatures.scala

```diff
@@ -44,7 +44,7 @@ object JavaSignatures:
               map.asInstanceOf[Map[TypeName, RegularSymbol]].get(tname) match
                 case Some(sym) => Some(TypeRef(NoPrefix, sym))
                 case None      => lookupTParam(member.outer)
-            case pt: Binders =>
+            case pt: TypeLambdaType =>
```

Comment:
this could remain `Binders` if `BoundType` was made to extend `Type`, then these lookup methods could be made abstract in `Binders`